designs/Net5PlatformAndPlatformVersion.md

```diff
@@ -0,0 +1,177 @@
+        {
+        }
+
+        private const int Version5 = 5;
+
+        public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, string profileOrPlatform, Version platformVersion)
+        {
+            if (frameworkIdentifier == null)
+            {
+                throw new ArgumentNullException("frameworkIdentifier");
+            }
+
+            if (frameworkVersion == null)
+            {
+                throw new ArgumentNullException("frameworkVersion");
+            }
+
+            _frameworkIdentifier = frameworkIdentifier;
+            _frameworkVersion = NormalizeVersion(frameworkVersion);
+
+            IsNet4Era = (_frameworkVersion.Major >= Version5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, _frameworkIdentifier));
+
+            _frameworkProfile = IsNet5Era ? profileOrPlatform ?? string.Empty : string.Empty;
+            Platform = IsNet5Era ? profileOrPlatform ?? string.Empty : string.Empty;
+            PlatformVersion = NormalizeVersion(platformVersion ?? FrameworkConstants.EmptyVersion);
+        }
+```
+
+### GetDotNetFrameworkName
+
+This method needs to be updated to append the Platform and PlatformVersion. I believe the format is supposed to be:
+
+```csharp
+string.Format(CultureInfo.InvariantCulture, "Platform={0},PlatformVersion={0}", framework.Platform, framework.PlatformVersion);
+```
+
+### GetShortFolderName
+
+This method needs to be updated such that when `IsNet5Era` is `true` and `Platform` is not `null`, `"-{Platform}[PlatformVersion]` is appended to the `net5.0` short tfm. This should be similar to how profiles are appended.
+
+## class NuGetFrameworkFactory
+
+Source: src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs
+
+### ParseFrameworkName
+
+This seems to be the core of the "long" tfm parsing logic. This needs to be modified to detect `Platform=` and `PlatformVersion=` and extract those accordingly. It should also skip parsing out `Profile` if it detects that the framework it's currently parsing is net5.0+. If an invalid Platform is detected, it should error out.
+
+### ParseFolder
+
+This seems to be the core of the "short" tfm parsing logic. As with `ParseFrameworkName`, this needs to be modified such that, when the identifier is net5.0+, it will check `FrameworkConstants.FrameworkPlatforms` and verify that the given platform is one of the accepted ones (per the net5.0 spec linked at top). If an invalid Platform is detected, it should error out.
+
+### RawParse
+
+This... mostly seems to do what I think it needs to do? It should probably have a big fat comment noting that we're piggybacking off profile parsing (keeping the net5.0+ dependent logic in `ParseFolder`). Maybe renaming variables to `profileOrPlatform` for clarity.
+
+## class FrameworkConstants
+
+Source: src/NuGet.Core/NuGet.Frameworks/FrameworkConstants.cs
+
+### (new!) FrameworkPlatforms

```

Comment:
Could we get these through dotnet api call instead?

Reply:
I asked the SDK team about this in our sync meeting, and consensus was that this should probably live in NuGet, but that errors might, actually, be a bit better if we moved it to the SDK. It makes implementation more complicated, potentially, because we'd have to wait on dotnet.